lib/web_ui/lib/src/engine/js_interop/js_app.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@JS()
+library js_app;
+
+import 'dart:js_interop';
+
+import 'package:ui/src/engine.dart';
+
+/// The JS bindings for the object that's set as `window.flutterConfiguration`.
+@JS()
+@anonymous
+@staticInterop
+class JsFlutterViewOptions {
+  external factory JsFlutterViewOptions();
+}
+
+/// The attributes of the [JsFlutterViewOptions] object.
+extension JsFlutterViewOptionsExtension on JsFlutterViewOptions {
+  external DomElement? get hostElement;
+
+  @JS('initialData')
+  external JSObject? get _initialData;
+  Object? get initialData => _initialData?.toObjectDeep;
```

Comment:
Do we need to define this?

If the user passes an `options` object to us, we will keep it intact, and return it back to them. So they are free to put whatever they want on it. We should only define the bits that **_we_** care about here.

Reply:
I think this is needed if we want to pass it to a Flutter app through a `ui_web` `getViewConfiguration(viewId)` API (but I'm not sure, we can remove it if it's not needed)